common/lib/xmodule/xmodule/js/spec/video/video_player_spec.js

```diff
@@ -333,6 +333,34 @@ function (VideoPlayer) {
                     expect($.fn.trigger).toHaveBeenCalledWith('ended', {});
                 });
             });
+
+            describe('emitPlayVideoEvent', function () {
+                beforeEach(function () {
+                    state = jasmine.initializePlayer();
+                    state.videoEl = $('video, iframe');
+                });
+
+                it('sets the emitPlayVideoEvent to correct value', function () {
+                    var verifyPlayVideoEventInfo = function (playerState, expectedEmitPlayVideoEventValue) {
+                        state.videoPlayer.onStateChange({data: playerState});
+                        expect(state.videoPlayer.lastPlayerState).toBe(playerState);
+                        expect(state.videoPlayer.emitPlayVideoEvent).toBe(expectedEmitPlayVideoEventValue);
+                    };
+
+                    //Whenever video player goes into BUFFERING state the next state is PLAYING and we always
+                    //emit play_video event in PLAYING state which is incorrect. So to avoid this, we will
+                    //control play_video event using emitPlayVideoEvent and lastPlayerState. In BUFFERING
+                    //state, check if lastPlayerState is PLAYING then set emitPlayVideoEvent to false and
+                    //then in PLAYING state play_video will not be emitted.
+
+                    verifyPlayVideoEventInfo(YT.PlayerState.PLAYING, true);
+                    verifyPlayVideoEventInfo(YT.PlayerState.BUFFERING, false);
+                    verifyPlayVideoEventInfo(YT.PlayerState.PLAYING, false);
+                    verifyPlayVideoEventInfo(YT.PlayerState.PAUSED, false);
```

Comment:
I don't think this is what we want. If the video was playing and is then paused and played again event should be triggered.